ush/gfs/task/analysis.py

```diff
@@ -0,0 +1,411 @@
+        import yaml
+        with open(self.berror_yaml, 'r') as yamlopen:
+            berror_yaml_dict = yaml.safe_load(yamlopen)
+
+    def stage_obs(self, filedict):
+        logging.info('Staging observations')
+        self.stage(filedict, skip_missing=True)
+        logging.info('Finished staging observations')
+
+    def stage_bkg(self, filedict):
+        logging.info('Staging model backgrounds')
+        self.stage(filedict)
+        logging.info('Finished staging backgrounds')
+
+    def stage_fix(self, filedict):
+        logging.info('Staging fix files')
+        self.stage(filedict)
+        logging.info('Finished staging fix files')
+
+    def stage_berror(self, filedict):
+        logging.info('Staging staticb files')
+        self.stage(filedict)
+        logging.info('Finished staging staticb files')
+
+    def stage(self, filedict, skip_missing=False):
+        for src, dest in filedict.items():
+            destdir = os.path.dirname(dest)
+            if not os.path.exists(destdir):
+                logging.info(f'{destdir} does not exist, creating directory.')
+                os.makedirs(destdir)
+            if os.path.exists(dest):
+                os.remove(dest)
+            if skip_missing:
+                if not os.path.exists(src):
+                    logging.warning(f'{src} does not exist. Will not copy.')
+                    continue
+            logging.info(f'Copying {src} to {dest}')
+            shutil.copyfile(src, dest)
+
+    def generate_yaml(self, extra_config={}):
+        """
+        Use existing tools to generate YAML from config and template
+        """
+        import copy
+        import ufsda # temporary until this is in workflow
+        output_yaml = f'{self.taskname}_{self.cdate}.yaml'
+        output_yaml_path = os.path.join(self.datadir, output_yaml)
+        template = self.yamltemplate
+        logging.info(f'Generating analysis YAML using {template}')
+        full_config = copy.deepcopy(self.config)
+        full_config.update(extra_config)
+        ufsda.yamltools.genYAML(full_config, template=template, output=output_yaml_path)
+        logging.info(f'Wrote YAML file to {output_yaml_path}')
+
+
+class AerosolAnalysis(Analysis):
+    """
+    Class for global aerosol analysis tasks
+    """
+    def __init__(self, config):
+        super().__init__(config)
+        self.yamltemplate = config['AEROVARYAML']
+        self.taskname = f'{self.cdump}aeroanl'
+
+    def initialize(self):
+        super().initialize()
+        logging.info('Initializing global aerosol analysis')
+        fix_dict = self.get_fix_file_dict()
+        self.stage_fix(fix_dict)
+        crtm_fix_dict = self.get_crtm_coeff_dict()
+        self.stage_crtm(crtm_fix_dict)
+        if os.environ.get('STATICB_TYPE', 'bump_aero') in ['bump_aero']:
+            self.stage_berror(self.get_berror_dict())
+        bkg_dict = self.get_bkg_dict()
+        self.stage_bkg(bkg_dict)
+        yaml_config = {
```

Comment:
Instead of hard-coding a bunch of things, can this just be another yaml file with templates that get filled in?

Reply:
Maybe? Will need to think about it, I don't see why not.